### ISSUE TITLE
Change argument "task.json" to "service.json"

### ISSUE THREAD

**Issue by JonRoma** [techservicesillinois/terraform-aws-ecs-service#15]
<a href="https://github.com/JonRoma"><img src="https://avatars2.githubusercontent.com/u/7817366?v=4" align="left" width="96" height="96" hspace="10"></img></a> **Issue by [JonRoma](https://github.com/JonRoma)**
_Wednesday Aug 29, 2018 at 23:41 GMT_
_Originally opened as https://github.com/cites-illinois/as-aws-modules/issues/78_

----

The ecs-service module defines a `container_definition_file` and then goes on to default that filename to `task.json` and refer to it as a task definition file:

> `container_definition_file` - (Optional) A file containing a list of valid container definitions provided as a single valid JSON document. Please note that you should only provide values that are part of the container definition document. For a detailed description of what parameters are available, see the Task Definition Parameters section from the official Developer Guide. The default filename is `task.json`.

Terraform's `ecs_task_definition` module allows the caller to define the task definition in native Terraform language (HCL), with the exception that the `container_definition` variable consists of raw JSON read from a file in the Terraform directory. 

After Terraform does its job, this is all passed to AWS as a _single_ JSON object – the task definition – of which, the container definition is a subset.

The terminology that we chose in `ecs-service` is potentially confusing, because the object described in the AWS manuals as a "task definition" is structurally different from the file that we give a default name of `task.json`.

Before we release this module out into the wild, it seems that we should eliminate this point of potential confusion. I suggest we change the default file name to `container.json` or `service.json` so that we don't inadvertently muddy the water with respect to a task definition (the superset) and container definition (the subset). ECS is confusing enough without tripping up our customers through ill-chosen nomenclature. 

The Terraform documentation has an example that looks like
```
container_definitions = "${file("task-definitions/service.json")}"
```

For this reason, perhaps `service.json` is better than `container.json`. The JSON _does_ define the service, though it defines it in the context of one or more containers. In short, either term is better than `task.json`.

Needless to say, this is not a complex change, but it will require attention to the example code as well as the documentation.

**Comment by JonRoma** [comment]
<a href="https://github.com/ddriddle"><img src="https://avatars2.githubusercontent.com/u/10728345?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [ddriddle](https://github.com/ddriddle)**
_Friday Aug 31, 2018 at 14:06 GMT_

----

@JonRoma I could live with `service.json` but I think `containers.json`would be clearer, or even I guess `container_definitions.json` but that is a bit wordy.

**Comment by JonRoma** [comment]
<a href="https://github.com/ddriddle"><img src="https://avatars2.githubusercontent.com/u/10728345?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [ddriddle](https://github.com/ddriddle)**
_Wednesday Sep 12, 2018 at 13:51 GMT_

----

@JonRoma I came up with another alternative name `container_defs.json`. Maybe we should get input from the team to see what they think is the best name.

**Comment by JonRoma** [comment]
<a href="https://github.com/JonRoma"><img src="https://avatars2.githubusercontent.com/u/7817366?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [JonRoma](https://github.com/JonRoma)**
_Wednesday Sep 12, 2018 at 18:40 GMT_

----

We agreed to use `containers.json` because that term is consistent with the Terraform variable `container_definitions` in referring to containers, and also in terms of being plural.

AWS now supports YAML in addition to JSON for the container definition. YAML would be our preferred default due to its improved readability. However, Terraform does not yet support, so JSON will remain the default for now.

**Comment by JonRoma** [comment]
Fixed in d10cd2e1773fbd643fd3d68a3665a696c6438061.
